pkg/registry/generic/rest/proxy.go

```diff
@@ -188,7 +209,11 @@ func (h *UpgradeAwareProxyHandler) dialURL() (net.Conn, error) {
 func (h *UpgradeAwareProxyHandler) defaultProxyTransport(url *url.URL) http.RoundTripper {
 	scheme := url.Scheme
 	host := url.Host
-	pathPrepend := strings.TrimRight(url.Path, h.Location.Path)
+	suffix := h.Location.Path
+	if strings.HasSuffix(url.Path, "/") && !strings.HasSuffix(suffix, "/") {
+		suffix += "/"
+	}
```

Comment:
if h.Location.Path=="/", then this will remove a trailing "/" from url.Path. Is that intentional?